src/rules/effects/prefer-effect-callback-in-block-statement.ts

```diff
@@ -26,19 +26,52 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
   },
   defaultOptions: [],
   create: (context) => {
+    const sourceCode = context.getSourceCode()
+    const nonParametrizedEffect =
+      `${createEffectExpression} > ArrowFunctionExpression > .body[type!=/^(ArrowFunctionExpression|BlockStatement)$/]` as const
+    const parametrizedEffect =
+      `${createEffectExpression} > ArrowFunctionExpression > ArrowFunctionExpression > .body[type!='BlockStatement']` as const
+    const parametrizedEffectWithinBlockStatement =
+      `${createEffectExpression} > ArrowFunctionExpression > BlockStatement > ReturnStatement > ArrowFunctionExpression > .body[type!='BlockStatement']` as const
+
     return {
-      [`${createEffectFunction} > CallExpression`](
-        node: TSESTree.CallExpression,
+      [`${nonParametrizedEffect}, ${parametrizedEffect}, ${parametrizedEffectWithinBlockStatement}`](
```

Comment:
I tried `${createEffectExpression} > :matches(${nonParametrizedEffect}, ${parametrizedEffect}, ${parametrizedEffectWithinBlockStatement})` to avoid repeating the `createEffectExpression`, but for some reason it didn't work 🤷‍♂️